### PR TITLE
loader: fix break in forloop and some minor refine

### PR DESCRIPTION
**Pull request by amyangfei**
### What problem does this PR solve? 

"break" statement terminates execution of the innermost "for", "switch" or "select" statement. We should cancel execution when `context.Done` is detected.

### What is changed and how it works?

1. fix a break wrong usage
2. remove some unused code
3. use a range operator to simplify single case in `for { select {} }`

### Check List 

Tests 

 - Unit test
 - Integration test

### CODE REVIEW

**Comment by amyangfei** [comment]
PTAL @GregoryIan @csuzhangxc

File: loader/loader.go [hunk 1]
```diff
@@ -1035,7 +1029,7 @@ func (l *Loader) restoreData(ctx context.Context) error {
 		select {
 		case <-ctx.Done():
 			log.Infof("stop dispatch data file job because %v", ctx.Err())
-			break
+			return nil
```
Comment (kennytm) [inline loader/loader.go]: No need to `l.closeFileJobQueue()`?

Comment (amyangfei) [reply loader/loader.go]: `l.closeFileJobQueue()` just set `fileJobQueueClosed` flag to `true` and close `fileJob` channel, in this scenario it is not needed, however calling it has no side effect.

We have not made the resource recycle regulation when we meet the context cancelation.
For one thing, if we always check the context before any resource access, we can always avoid invalid resource access. On the other side, active close channel, reset resource to nil is always ok.

**Comment by WangXiangUSTC** [comment]
LGTM

**Pull request merged**